src/db.c

```diff
@@ -597,10 +597,9 @@ void existsCommand(client *c) {
 }
 
 void selectCommand(client *c) {
-    long id;
+    int id;
 
-    if (getLongFromObjectOrReply(c, c->argv[1], &id,
-        "invalid DB index") != C_OK)
+    if (getIntFromObjectOrReply(c, c->argv[1], &id, NULL) != C_OK)
```

Comment:
I am not sure we should do that.

For error message, I think we should respect before.

Reply:
I think the previous error message was wrong, if I pass a non-number and the error is "invalid DB index", I think it should explicitly tell the non-number.

Reply:
Yes, it seems you are right

Reply:
the changes to the error messages are in some way breaking backwards compatibility, so although they are better, we might want to avoid that.

on the other hand, these errors are on invalid use, not a runtime error like type mismatch, so maybe there's no application that actually relies on matching these strings.

Reply:
I think as long as we emit the same error type, the actual wording is okay.

Reply:
@oranagra I think the original error message didn't represent the error message very well, and while it's not backwards compatible, I think it's worth doing in the long run, and is much more user friendly!